### PR TITLE
Sample for testing Teams Tabs

### PR DESCRIPTION
**Pull request by axelsrz**
Fixes: Our lack of this sample in python.

### CODE REVIEW

**Comment by bhargavimohan** [comment]
I tried to run this after installing the SDK from `https://github.com/microsoft/botbuilder-python/tree/axsuarez/teams-ac-tabs` 

but i seem to get this error

`ImportError: cannot import name 'TabResponse'`

TabResponse also does not seem to be defined in botbuilder-schema

**Comment by EricDahlvang** [comment]
Hi @bhargavimohan 

We've fixed a bug related to this:
https://github.com/microsoft/botbuilder-python/pull/1538

As well as some updates to the sample:
https://github.com/microsoft/BotBuilder-Samples/pull/3087/commits/d42d9370e4115ec50676ece0403698eaa70e137d

File: experimental/teams-tabs/python/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Teams Task Module
```
Comment (mdrichardson) [inline experimental/teams-tabs/python/README.md:1]: This should reflect Teams Tabs. [see here](https://github.com/mdrichardson/BotBuilder-Samples/blob/teamsTabs/experimental/teams-tabs/javascript_nodejs/README.md)

File: experimental/teams-tabs/python/bots/teams_adaptive_card_tabs_bot.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    UIConstants,
+    TaskModuleResponseFactory,
+)
+
+from simple_graph_client import SimpleGraphClient
+
+
+class TeamsAdaptiveCardTabsBot(TeamsActivityHandler):
+    def __init__(self, config: DefaultConfig):
+        self.__base_url = config.BASE_URL
```
Comment (mdrichardson) [inline experimental/teams-tabs/python/bots/teams_adaptive_card_tabs_bot.py:41]: I believe this is unused and can be omitted.

**Review by mdrichardson — CHANGES_REQUESTED** [review]
Just a couple nit-picky things. My review also assumes this works--I can't test it, myself, due to not being able to get on the right Teams ring.

**Comment by bhargavimohan** [comment]
Hi Eric,

Thank you for your email. I am looking for an example for configurable tabs
in python. Can you please provide me with an example sample.

I am trying to design a tab that renders the content from my ngrok
endpoint. I tried to add a configurable tab but i get the following error
marked in a rectangle. Or can you hint me with what I am missing? It would
be of great help

[image: image.png]

Thank you
Bhargavi Mohan

**Comment by EricDahlvang** [comment]
I'm sorry @bhargavimohan but, this is a Draft Pull Request into the BotBuilder-Samples repository.  Not a valid location for question/answers.  Please direct questions to Stack Overflow.

I'm not sure about Teams 'configurable tabs'.  This PR is related to a new feature: Adaptive Card Tabs.

**Comment by bhargavimohan** [comment]
Hey!

Thanks for your quick response. I understand that.

I figured out about configuration tabs! Thanks anyway :)

**Comment by EricDahlvang** [comment]
@bhargavimohan  good to hear :)

File: experimental/teams-tabs/python/deploymentTemplates/template-with-new-rg.json [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                                    }
+                                }
+                            }
+                        },
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
+                            "location": "global",
+                            "kind": "bot",
```
Comment (BruceHaley) [inline experimental/teams-tabs/python/deploymentTemplates/template-with-new-rg.json:159]: "kind": "bot" is out of date, no longer supported. See my earlier comment.

File: experimental/teams-tabs/python/deploymentTemplates/template-with-preexisting-rg.json [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                    }
+                }
+            }
+        },
+      {
+        "apiVersion": "2017-12-01",
+        "type": "Microsoft.BotService/botServices",
+        "name": "[parameters('botId')]",
+        "location": "global",
+        "kind": "bot",
```
Comment (BruceHaley) [inline experimental/teams-tabs/python/deploymentTemplates/template-with-preexisting-rg.json:135]: "kind": "bot" is out of date, no longer supported. See my earlier comment.

**Review by BruceHaley — CHANGES_REQUESTED** [review]

**Comment by EricDahlvang** [comment]
Closing, we've decided to not include this.

**Pull request closed without merging**